### PR TITLE
Operation Based projection unit test generation

### PR DESCRIPTION
**Pull request by hamzajg**

### CODE REVIEW

**Review by VaughnVernon — APPROVED** [review]
@hamzajg Good, thanks. Need review from @danilo-ambrosio before merge.

**Comment by danilo-ambrosio** [comment]
> @hamzajg Good, thanks. Need review from @danilo-ambrosio before merge.

Will review it today @hamzajg . Thanks!

**Review by danilo-ambrosio — APPROVED** [review]

**Pull request merged**
